gallery/__init__.py

```diff
@@ -772,7 +773,12 @@ def display_thumbnail(file_id: int, auth_dict: Optional[Dict[str, Any]] = None):
     file_model = File.query.filter(File.id == file_id).first()
 
     link = storage_interface.get_link("thumbnails/{}".format(file_model.s3_id))
-    return redirect(link)
+    if "LOCAL_STORAGE_PATH" in app.config:
+        link = "http://" + app.config["SERVER_NAME"] + link
```

Comment:
Is this doing something that get_link isn't?

Reply:
get_link for local only returns `/public/.………..`, which requests doesn't understand.

Reply:
Looking back I could just mess with the storage util file